test/sets.jl

```diff
@@ -810,6 +822,19 @@ end
     @test replace((NaN, 1.0), NaN=>0.0) === (0.0, 1.0)
     @test replace([1, missing], missing=>0) == [1, 0]
     @test replace((1, missing), missing=>0) === (1, 0)
+
+    # test replace and friends for AbstractDicts
+    d1 = ReplaceDict(Dict(1=>2, 3=>4))
+    d2 = replace(d1, (1=>2) => (1=>"a"))
+    @test d2 == ReplaceDict(Dict(1=>"a", 3=>4))
+    replace!(d1, (1=>2) => (1=>-2))
+    @test d1 == ReplaceDict(Dict(1=>-2, 3=>4))
+
+    d1 = ReplaceDict(Dict(1=>2, 3=>1, 5=>1, 7=>1))
+    @test replace(d1, (1=>2) => (1=>"a"), count=0) == d1
+    d2 = replace((kv)->(kv[2] == 1 ? kv[1]=>2 : kv), d1, count=2)
+    @test count(v->v==2, values(d2)) == 3
+    @test count(v->v==1, values(d2)) == 1
```

Comment:
It's subjective comment (feel free to ignore), but it's hard to follow whether this property was already tested for `Dict`. If it was not, I would try to make this test for both types, e.g. : 
```suggestion
    dd = Dict(1=>2, 3=>1, 5=>1, 7=>1)
    for d1 = (dd, ReplaceDict(dd))
        @test replace(d1, (1=>2) => (1=>"a"), count=0) == d1
        d2 = replace((kv)->(kv[2] == 1 ? kv[1]=>2 : kv), d1, count=2)
        @test count(v->v==2, values(d2)) == 3
        @test count(v->v==1, values(d2)) == 1
    end
```
It it was already tested, i would similarly update the existing test to also test `ReplaceDict`.